packages/gestalt-codemods/46.0.0/replace-typography-sizes.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Converts
+ *   size="sm"|"md"|"lg"
+ * to
+ *   size="100"|"200"|"300"
+ * for Text elements
+ * and
+ * Converts
+ *   size="sm"|"md"|"lg"
+ * to
+ *   size="400"|"500"|"600"
+ * for Heading elements
+ */
+
+// yarn codemod --parser=flow -t=packages/gestalt-codemods/46.0.0/replace-typography-sizes.js relative/path/to/your/code
+
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const src = j(file.source);
+  const TEXT_MAPPING = {
+    'sm': '100',
+    'md': '200',
+    'lg': '300',
+  };
+  const HEADING_MAPPING = {
+    'sm': '400',
+    'md': '500',
+    'lg': '600',
+  };
+  const CURRENT_SIZE_VALUES = ['sm', 'md', 'lg'];
+
+  let localIdentifierNames;
+
+  src.find(j.ImportDeclaration).forEach((path) => {
+    const decl = path.node;
+    if (decl.source.value !== 'gestalt') {
+      return null;
+    }
+
+    localIdentifierNames = decl.specifiers
+      .filter((node) => ['Heading', 'Text'].includes(node.imported?.name))
+      .map((node) => node.local?.name);
+
+    return null;
+  });
+
+  if (!localIdentifierNames || localIdentifierNames.length === 0) {
+    return null;
+  }
+
+  return src
+    .find(j.JSXElement)
+    .forEach((jsxElement) => {
+      const { node } = jsxElement;
+
+      if (!localIdentifierNames.includes(node.openingElement.name.name)) {
+        return null;
+      }
+
+      const attrs = node.openingElement.attributes;
+
+      if (attrs.some((attr) => attr.type === 'JSXSpreadAttribute')) {
+        throw new Error(
+          `Remove Dynamic Heading and Text properties and rerun codemod. Location: ${file.path} @line: ${node.loc.start.line}`,
+        );
+      }
+
+      const isHeadingComponent = node.openingElement.name.name.includes('Heading');
+
+      node.openingElement.attributes = attrs
+        .map((attr) => {
+          if (attr?.name?.name === 'size' && attr.value.type === 'JSXExpressionContainer') {
+            if (
```

Comment:
It's difficult to tell in the GH UI, but I _think_ this will still miss something like `size={condition ? 'sm' : foo}`. The way I'm reading this is that the old value will be transformed, but there will be no alert about the `foo` variable, which will almost certainly point to old values. Can you add test cases for this specific scenario to make sure the codemod works as expected?